cmd/restore.go

```diff
@@ -142,17 +159,45 @@ var restoreCmd = &cobra.Command{
 	//******************************************************************************************
 
 	PersistentPostRunE: func(cmd *cobra.Command, args []string) (err error) {
-		client, ok := cmd.Context().Value(keys.CLIENT_CONTEXT_KEY).(*client.Client)
-		if !ok {
-			return fmt.Errorf("invalid client in context")
-		}
-		defer client.Close()
-
 		// Assuming request is now ready to be sent to the server
 		req, ok := cmd.Context().Value(keys.RESTORE_REQ_CONTEXT_KEY).(*daemon.RestoreReq)
 		if !ok {
 			return fmt.Errorf("invalid restore request in context")
 		}
+		client, ok := cmd.Context().Value(keys.CLIENT_CONTEXT_KEY).(*client.Client)
+		if !ok {
```

Comment:
Please flip this if condition, keeping smaller code (for server-based restore) inside the if. This is a bit unintuitive.